src/models/WidgetTemplateImport.php

```diff
@@ -0,0 +1,212 @@
+<?php
+
+namespace hrzg\widget\models;
+
+use hrzg\widget\helpers\WidgetTemplateExport;
+use hrzg\widget\models\crud\WidgetTemplate;
+use hrzg\widget\widgets\TwigTemplate;
+use yii\base\ErrorException;
+use yii\base\InvalidArgumentException;
+use yii\base\InvalidConfigException;
+use yii\base\Model;
+use yii\helpers\FileHelper;
+use yii\helpers\Inflector;
+use yii\helpers\Json;
+
+/**
+ * --- PROPERTIES ---
+ *
+ * @property \yii\web\UploadedFile[] $tarFiles
+ *
+ * @author Elias Luhr
+ */
+class WidgetTemplateImport extends Model
+{
+    /**
+     * @var \yii\web\UploadedFile[]
+     */
+    public $tarFiles;
+
+    /**
+     * Import extract directory of the generated tar file
+     * If this is set with an alias, it will be automatically resolved later on
+     *
+     * @var string
+     */
+    public $importDirectory = '@runtime/tmp/dmstr/widgets/templates/import';
+
+    /**
+     * @return void
+     * @throws \yii\base\Exception if the export directory could not be created due to an php error
+     * @throws \yii\base\InvalidConfigException if either the import directory is not set or the export directory cannot
+     * be created due to permission errors
+     */
+    public function init()
+    {
+        parent::init();
+
+        // Ensure that the import directory for the generated files is set
+        if (empty($this->importDirectory)) {
+            throw new InvalidConfigException('$importDirectory must be set');
+        }
+
+        // Make sure that if an alias is set, it is resolved correctly
+        $this->importDirectory = \Yii::getAlias($this->importDirectory);
```

Comment:
directories (and/or filenames) used for imports should always have some randomness to prevent race conditions.